client/x/evmstaking/keeper/deposit.go

```diff
@@ -96,8 +99,78 @@ func (k Keeper) ProcessDeposit(ctx context.Context, ev *bindings.IPTokenStakingD
 	}
 
 	val, err := k.stakingKeeper.GetValidator(cachedCtx, validatorAddr)
+
+	//nolint:nestif // nested ifs error handling
 	if errors.Is(err, stypes.ErrNoValidatorFound) {
-		return errors.WrapErrWithCode(errors.ValidatorNotFound, err)
+		log.Debug(cachedCtx, "Validator not found, refunding deposit minus the refund fee",
+			"val_story", validatorAddr,
+			"val_pubkey", validatorPubkey.String(),
+		)
+
+		// the min refund fee amount will be `refundFeeBps * minDelegationAmount (1024) / 10_000bps`
+		refundFeeBps, err := k.RefundFeeBps(cachedCtx)
+		if err != nil {
+			return errors.Wrap(err, "get refund fee")
+		}
+		refundFeeAmount := amountCoin.Amount.Mul(math.NewInt(int64(refundFeeBps))).Quo(math.NewInt(10_000))
+		refundAmount := amountCoin.Amount.Sub(refundFeeAmount)
+
+		// refund completion time is 1 day after the deposit
+		completionTime := sdkCtx.BlockTime().Add(time.Hour * 24)
+
+		defer func() {
```

Comment:
Can we combine this defer statement into above one?

It would be good to add new error codes, where the error codes change depending on whether a refund occurs. If a refund occurs, additional event attributes can be included and emitted. Since a refund is also a type of delegation failure case, the event type should be `EventTypeDelegateFailure`.

Reply:
combined in d96d7db